File: Easy/27-remove-element/try-3.cpp
/*
 * Problem #27: Remove Element
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 02/10/2024, 12:41:35
 * Link: https://leetcode.com/problems/remove-element/
 */

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int h=0,i;
        for(i=0;i<nums.size();i++){
            if(nums[i]!=val){
                nums[h]=nums[i];
                h++;
            }
        }
        return h;
        
    }

};

File: Medium/215-kth-largest-element-in-an-array/try-3.py
/*
 * Problem #215: Kth Largest Element in an Array
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 16/12/2024, 18:59:17
 * Link: https://leetcode.com/problems/kth-largest-element-in-an-array/
 */

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap=[-x for x in nums]
        heapify(heap)
        for i in range(k-1):
            heappop(heap)
        return -heappop(heap)
        
